Ginger/Ginger/ALM/Repository/QtestRepository.cs

```diff
@@ -0,0 +1,405 @@
+using TDAPIOLELib;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Repository;
+using GingerCoreNET.SolutionRepositoryLib.RepositoryObjectsLib.PlatformsLib;
+using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common.InterfacesLib;
+using GingerCore.ALM.Qtest;
+using Ginger.ALM.Qtest.TreeViewItems;
+
+namespace Ginger.ALM.Repository
+{
+    class QtestRepository : ALMRepository
+    {
+        QtestTest matchingTC = null;
+        public override bool ConnectALMServer(ALMIntegration.eALMConnectType userMsgStyle)
+        {
+            try
+            {
+                Reporter.ToLog(eLogLevel.DEBUG, "Connecting to QC server");
+                return ALMIntegration.Instance.AlmCore.ConnectALMServer();
+            }
+            catch (Exception e)
+            {
+                if (userMsgStyle == ALMIntegration.eALMConnectType.Manual)
+                    Reporter.ToUser(eUserMsgKey.QcConnectFailure, e.Message);
+                else if (userMsgStyle == ALMIntegration.eALMConnectType.Auto)
+                    Reporter.ToUser(eUserMsgKey.ALMConnectFailureWithCurrSettings, e.Message);
+
+                Reporter.ToLog(eLogLevel.WARN, "Error connecting to QC server", e);
+                return false;
+            }
+        }
+
+        #region General
+        public override IEnumerable<Object> SelectALMTestSets()
+        {
+            //show Test Lab browser for selecting the Test Set/s
+            QCTestLabExplorerPage win = new QCTestLabExplorerPage(QCTestLabExplorerPage.eExplorerTestLabPageUsageType.Select);
+            return (ObservableList<QCTestSetTreeItem>)win.ShowAsWindow(eWindowShowStyle.Dialog);
+        }
+
+        public override string SelectALMTestPlanPath()
+        {
+            //show Test Cycles browser for selecting the Path
+            QtestCyclesExplorerPage win = new QtestCyclesExplorerPage(string.Empty, true);
+            object selectedPathObject = win.ShowAsWindow(eWindowShowStyle.Dialog);
+            if (selectedPathObject is QtestSuiteTreeItem)
+            {
+                return ((QtestSuiteTreeItem)selectedPathObject).ID.ToString();
+            }
+            if (selectedPathObject is QtestCycleTreeItem)
+            {
+                return ((QtestCycleTreeItem)selectedPathObject).ID.ToString();
+            }
+            return string.Empty;
+        }
+
+        public override string SelectALMTestLabPath()
+        {
+            //show Test Lab browser for selecting the Test Set/s
+            QCTestLabExplorerPage win = new QCTestLabExplorerPage(QCTestLabExplorerPage.eExplorerTestLabPageUsageType.BrowseFolders);
+            return (string)win.ShowAsWindow(eWindowShowStyle.Dialog);
+        }
+
+        public override List<string> GetTestLabExplorer(string path)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override IEnumerable<Object> GetTestSetExplorer(string path)
+        {
+            throw new NotImplementedException();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)